### DISCUSSION TITLE
[Newby] Custom Snippets non functional

### DISCUSSION THREAD

**Discussion by CarrollDiehl** [pulsar-edit/.github#247] [Pulsar Support]
Hey everyone, I hope this is the right place to ask.

I'm just writing on my thesis using language-latex and latextools in Pulsar.
First of all: It's amazing! So modern, functional, slim - I love it!

But I have some problems adding customized snippets.
Here an example: I made a simple `\todo{}` command in latex, which I want to create a snippet for.
Here is the content from my `snippets.cson`:

```
'.text.tex.latex':
  'todo':
    'prefix': '\todo'
    'body': '\todo{$1}'
```

When I go to my `.tex` document, it doesn't appear when typing `\todo`. 

`Alt+Shift+S`, which should show all applicable snippets (?), doesn't do anything.

The scope is the same scope the snippets in latextools use, and these work.

Maybe I am just blatantly missing something?
Thanks for any help (:

Best
Carroll

**Comment by savetheclocktower** [comment]
It's because the `\t` is being interpreted as a tab character. You need to escape it like this in both the prefix and the body:

```coffeescript
'.text.tex.latex':
  'todo':
    'prefix': '\\todo'
    'body': '\\todo{$1}'
```

That said, you've saved no keystrokes over the alternative of typing `\todo` and pressing `{` rather than `Tab` (since `{` and `}` are a smart typing pair). Aim higher! Save yourself more time! :-)

**Comment by CarrollDiehl** [comment]
Thank you so much!

Unfortunately, I have to inform you, that your proposed escape didn't work for me, too. They still didn't come up when typed. But I realized, that I could just change it to this:

```
'.text.tex.latex':
  'todo':
    'prefix': 'td'
    'body': 'todo{$1}'
```
And, as you can see, I saved two keystrokes now (: And many more, since I now know that my snippets are working.

Best regards
Carroll
